File: app/src/main/java/com/abrsoftware/myapplication/view/home/HomeMvp.java
package com.abrsoftware.myapplication.view.home;

import com.abrsoftware.myapplication.event.GeneralEvent;

public class HomeMvp {
    public interface View {
        void showLoading(boolean show);

        void showMsj(String msj);

        <T> void successResponce(T responce);
    }

    public interface Presenter {
        void oncreate();

        void onDestroy();

        void onEventMainThread(GeneralEvent event);

        void getRecommendation(String page, String numMoview);

        void getPolular(String userId);

        void getLastest(String userId);

        void getUpcomming(String userId);
    }

    public interface Repository {
        void getRecommendation(String page, String numMoview);

        void getPolular(String userId);

        void getLastest(String userId);

        void getUpcomming(String userId);
    }
}

File: app/src/main/java/com/abrsoftware/myapplication/view/home/HomeMvp.kt
package com.abrsoftware.myapplication.view.home

import com.abrsoftware.myapplication.event.GeneralEvent

class HomeMvp {
    interface View {
        fun showLoading(show: Boolean)
        fun showMsj(msj: String?)
        fun <T> successResponce(responce: T)
    }

    interface Presenter {
        fun oncreate()
        fun onDestroy()
        fun onEventMainThread(event: GeneralEvent?)
        fun getPolular(page: String?)
        fun getLastest(page: String?)
        fun getUpcomming(page: String?)
    }

    interface Repository {
        fun getPolular(page: String?)
        fun getLastest(page: String?)
        fun getUpcomming(page: String?)
    }
}